wgpu-hal/src/gles/egl.rs

```diff
@@ -757,13 +756,30 @@ impl crate::Surface<super::Api> for Surface {
             );
         }
 
-        let format_desc = device.shared.describe_texture_format(config.format);
         let gl = &device.shared.context;
+
+        // Check renderer version to see if we need to workaround intel mesa bug
+        if self.enable_intel_mesa_fastclear_workaround.is_none() {
+            let renderer = gl.get_parameter_string(glow::RENDERER);
+
+            // Check renderer string for buggy cards
+            self.enable_intel_mesa_fastclear_workaround = Some(
+                renderer.contains("Mesa")
+                    && renderer.contains("Intel")
+                    && renderer.contains("CML GT2"),
+            );
+        }
+
+        let format_desc = device.shared.describe_texture_format(config.format);
         let renderbuffer = gl.create_renderbuffer().unwrap();
         gl.bind_renderbuffer(glow::RENDERBUFFER, Some(renderbuffer));
         gl.renderbuffer_storage(
             glow::RENDERBUFFER,
-            format_desc.internal,
+            if self.enable_intel_mesa_fastclear_workaround.unwrap() {
+                glow::RGBA8
```

Comment:
What end result does this have? Why can we just use a different format?

Reply:
Does this actually start making the tests pass, considering this is just for the surface?

Reply:
> What end result does this have? Why can we just use a different format?

It fixes he pixelated edges on my machine. I don't know why, though. It may be related to how in the mesa issue the original poster said that if he commented out `glfwWindowHint(GLFW_SRGB_CAPABLE, 1);` it would render fine. It might be an issue specifically with the sRGB formats.

Beats me. 🤷‍♂️

---

> Does this actually start making the tests pass, considering this is just for the surface?

That's a good point. It doesn't fix the tests.

I don't know how it's fixing it for the window display, but if it isn't fixing it for the reftests then this probably isn't going to work.